aten/src/ATen/native/sparse/SparseBinaryOpIntersectionCommon.h

```diff
@@ -365,7 +342,7 @@ void _sparse_binary_op_intersection_kernel_impl(
   Tensor intersection_count, intersection_first_idx;
   std::tie(intersection_count, intersection_first_idx) = [&]() -> std::tuple<Tensor, Tensor> {
     const auto source_nnz = source._nnz();
-    auto intersection_buffer = at::empty({2, source_nnz}, sorted_hash.options());
+    auto intersection_buffer = at::empty({2, source_nnz}, sorted_hash.options().dtype(kLong));
```

Comment:
https://github.com/pytorch/pytorch/pull/92976/files#r1101848098 follows:
```suggestion
    auto intersection_buffer = at::empty({2, source_nnz}, sorted_hash.options());
```